app/models/post.rb

```diff
@@ -0,0 +1,24 @@
+class Post < ApplicationRecord
+  # Associations
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  # Attributes
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_couter, :integer, default: 0
+  attribute :likes_counter, :integer, default: 0
+
+  # Callback
+  after_save :update_user_posts_counter
+
+  # Methods
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count + 1)
+  end
+
+  def five_most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```

Comment:
- [OPTIONAL] "Great job on your `Post` model! Just a quick thought: consider marking the `update_user_posts_counter` method as private. This would help reinforce the idea that it's meant for internal use within the class and not intended to be called externally. Keep up the excellent work!"

```suggestion
class Post < ApplicationRecord
  # Associations
  belongs_to :author, class_name: 'User'
  has_many :comments
  has_many :likes

  # Attributes
  attribute :title, :string
  attribute :text, :text
  attribute :comments_couter, :integer, default: 0
  attribute :likes_counter, :integer, default: 0

  # Callback
  after_save :update_user_posts_counter

def five_most_recent_comments
    comments.order(created_at: :desc).limit(5)
  end
  
  private
  
  # Methods
  def update_user_posts_counter
    author.update(posts_counter: author.posts.count + 1)
  end
end
```